File: JJune/Level.1/정수_제곱근_판별.cpp
#include <string>
#include <vector>
#include <math.h>

using namespace std;

long long solution(long long n)
{
    uint64_t val = 1;
    while (val <= n)
    {
        if (pow(val, 2) != n)
        {
            val++;
        }
        else
        {
            return pow((val + 1), 2);
        }
    }
    return -1;
}

#include <iostream>
int main()
{
    std::cout << "Out : " << solution(121) << std::endl;
    std::cout << "Out : " << solution(3) << std::endl;
    return 0;
}

File: kkim218/Level.1/정수_제곱근_판별.java
class Solution {
  public long solution(long n) {
      long sqrt = (int)Math.sqrt(n); // 제곱근 값을 계산 ex) 121의 제곱근은 11 
      
      if (Math.pow(sqrt, 2) == n) // 거듭제곱 값을 계산 pow([대상 숫자], [거듭제곱 횟수])
          return (long)Math.pow((sqrt + 1), 2);
            
      return -1;
  }
}